src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -53,13 +53,96 @@ object FunctionBuilder {
 
   def functionBuilder[A: TypeInfo, B: TypeInfo, C: TypeInfo, D: TypeInfo, E: TypeInfo, R: TypeInfo]: FunctionBuilder[AsmFunction5[A, B, C, D, E, R]] =
     new FunctionBuilder(Array(GenericTypeInfo[A], GenericTypeInfo[B], GenericTypeInfo[C], GenericTypeInfo[D], GenericTypeInfo[E]), GenericTypeInfo[R])
+}
+
+class MethodBuilder(val fb: FunctionBuilder[_], mname: String, parameterTypeInfo: Array[TypeInfo[_]], returnTypeInfo: TypeInfo[_]) {
```

Comment:
MethodBuilder takes a FunctionBuilder?  Ah... where is this method being built?

Reply:
On the function builder's class. If we split things out into method builder and class builder, it would take a class builder.